internal/pkg/experimental/action/catalog_inspect.go

```diff
@@ -0,0 +1,160 @@
+package action
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/spf13/pflag"
+
+	"github.com/operator-framework/kubectl-operator/internal/pkg/catalogd/fetcher"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+)
+
+// CatalogInspect is a helper struct that implements
+// functionality to fetch catalog contents for the
+// "catalog inspect" subcommand
+type CatalogInspect struct {
+	config             *action.Configuration
+	Logf               func(string, ...interface{})
+	createFetcherFunc  func(*action.Configuration) CatalogFetcher
+	createStreamerFunc func(*action.Configuration) CatalogContentStreamer
+}
+
+// CatalogInspectOptions is the set of configurable
+// options that are used to filter the set of catalog
+// contents to return
+type CatalogInspectOptions struct {
+	// Catalog is the name of the catalog
+	// all returned contents should be from.
+	// Optional.
+	Catalog string
+	// Schema is the schema that all the returned
+	// contents should have.
+	// Optional.
+	Schema string
+	// Package is the package that all the returned
+	// contents should be from.
+	// Optional.
+	Package string
+	// Output is the format that contents should be
+	// written to STDOUT in.
+	// Optional.
+	Output string
+
+	// Name is the name of the catalog object to inspect. Required.
+	Name string
+}
+
+const OutputFlag = "output"
+
+// BindCatalogInspectFlags binds the flags for the "catalog inspect"
+// subcommand to the provided pflag.FlagSet
+func BindCatalogInspectFlags(flagSet *pflag.FlagSet) {
+	flagSet.String(CatalogFlag, "", "filter results to only be from the specified catalog")
+	flagSet.String(SchemaFlag, "", "filter results to only be FBC objects that have the specified schema")
+	flagSet.String(PackageFlag, "", "filter results to only be FBC objects that belong to the specified package")
+	flagSet.String(OutputFlag, "json", "the format in which output should be. One of [json, yaml]")
+}
+
+// CatalogInspectOptionsFromFlags will return a CatalogInspectOptions pointer with the options
+// fetched from the values for the corresponding flags in a given pflag.FlagSet. Returns nil and
+// an error if any errors are encountered during the process
+func CatalogInspectOptionsFromFlags(flagSet *pflag.FlagSet) (*CatalogInspectOptions, error) {
+	catalog, err := flagSet.GetString(CatalogFlag)
+	if err != nil {
+		return nil, fmt.Errorf("parsing catalog flag: %w", err)
+	}
+
+	schema, err := flagSet.GetString(SchemaFlag)
+	if err != nil {
+		return nil, fmt.Errorf("parsing schema flag: %w", err)
+	}
+
+	pack, err := flagSet.GetString(PackageFlag)
+	if err != nil {
+		return nil, fmt.Errorf("parsing package flag: %w", err)
+	}
+
+	output, err := flagSet.GetString(OutputFlag)
+	if err != nil {
+		return nil, fmt.Errorf("parsing output flag: %w", err)
+	}
```

Comment:
I've never been a fan of this style of flag value fetching, because it introduces a bunch of unnecessary indirection and error checking around the flag name being registered.

I also don't think it makes sense to pollute the `action` package with a bunch of CLI-related features.

My overall suggestion would be:
- Use `flagSet.StringVar*()`-style functions to bind flags directly to existing variables/struct fields
- Move all flag binding code into the cmd package.
- Potentially remove the options struct entirely and put the relevant fields directly on the actions struct.
- Make the `Output` field separate. That is a CLI concern only, right?

Reply:
I'm also not a huge fan of this approach and originally had the `flagSet.StringVar*()` style functions. Will move some stuff around to get rid of this.

Reply:
I just puushed up some changes. I was originally trying to avoid the use of global variables for flags but I am not aware of any other way to achieve this for the time being (I had tried within the function that creates the cobra command but that didn't seem to work, although  I could have been doing something wrong). 

I'm glad you pointed out the flaws with this previous approach. I did keep the options struct as I personally prefer that pattern over requiring it to be on the actions struct.